lib/aiken/rats.ak

```diff
@@ -0,0 +1,88 @@
+// Rationals
+// Authors:
+// Micah Kendall
+// <other authors go here>
+
+pub type UncheckedRational {
+  numerator: Int,
+  denominator: Int,
+}
+
+pub opaque type Rational {
+  numerator: Int,
+  denominator: Int,
+}
+
+pub fn check_rational(num: UncheckedRational) -> Rational {
+  let UncheckedRational { numerator, denominator } =
+    num
+  div_int(numerator, denominator)
+}
+
+pub fn mul(a: Rational, b: Rational) -> Rational {
+  Rational {
+    numerator: a.numerator * b.numerator,
+    denominator: a.denominator * b.denominator,
+  }
+}
+
+pub fn div(a: Rational, b: Rational) -> Rational {
+  Rational {
+    numerator: a.numerator * b.denominator,
+    denominator: a.denominator * b.numerator,
+  }
+}
+
+pub fn div_int(numerator: Int, denominator: Int) -> Rational {
+  if denominator < 0 {
+    Rational { numerator: -numerator, denominator: -denominator }
+  } else {
+    Rational { numerator, denominator }
+  }
+}
+
+pub fn add(a: Rational, b: Rational) -> Rational {
+  Rational {
+    numerator: a.numerator * b.denominator + b.numerator * a.denominator,
+    denominator: a.denominator * b.denominator,
+  }
+}
+
+pub fn sub(a: Rational, b: Rational) -> Rational {
+  Rational {
+    numerator: a.numerator * b.denominator - b.numerator * a.denominator,
+    denominator: a.denominator * b.denominator,
+  }
+}
+
+pub fn gt(a: Rational, b: Rational) -> Bool {
```

Comment:
Let's stick to "compare" here (and other related places) and return an `Ordering` instead. So that it's consistent with the overall stdlib

Reply:
Additionally compare perhaps but I think these should be kept because compare is unwieldy.

Reply:
I also agree that `Ordering` is pretty unwieldy. I actually think using `<,>,==,...` is more consistent with the overall stdlib since the `Rational` type is meant to be treated as a number like `Int`. Since `Int` supports these operators, I never find myself using `compare` with them even though it is supported. I'm not sure which is more performant though.

Reply:
What I'd really like ultimately is to support some form of operator overloading and stick to `compare` until we have full blown traits / type-classes. So that we get the best of both worlds.

Reply:
Until then, this shorthand with bools is what we have. We could open an issue to update this library as soon as overloading comes out. These functions are meant to be the overloaded functions, there is just no mechanism for that, so leaving them as they are is best imo.